src/Common/src/System/Security/Cryptography/DSACng.cs

```diff
@@ -14,11 +15,13 @@ internal static partial class DSAImplementation
 #endif
         public sealed partial class DSACng : DSA
         {
+            private static readonly int s_defaultKeySize = Supports2048KeySize() ? 2048 : 1024;
```

Comment:
I'm torn on whether we should adjust s_defaultKeySize based on that input as well (since Win7 is capped at 1024).  But I don't see any evidence of us ever making LegalKeySizes return different answers on different OSes.